[BOT] Rupato/bot 1011/atler rudderstack implementation bot

rupato-deriv opened the pull request:
Altered the rudderstack implemetation for bot

review-deriv reviewed (commented):
PR description is too short. You can use this format:

```
# Description
This is a summary of the changes. This should describe why do we need this change.
Whys and Hows come here.

# Cards
* https://clickup.com/12345
* https://clickup.com/12346

```

NikitK-deriv commented on packages/bot-web-ui/src/components/quick-strategy/inputs/qs-input.tsx:
```diff
@@ -23,6 +26,38 @@ const QSInput: React.FC<TQSInput> = observer(
+            });
+        };
+
+        const throttleChangePlus = React.useCallback(
+            debounce(sendMinusValueToRudderstack, DEBOUNCE_INTERVAL_TIME, {
+                trailing: true,
+                leading: false,
+            }),
+            []
+        );
+
+        const throttleChangeMinus = React.useCallback(
+            debounce(sendPlusValueToRudderstack, DEBOUNCE_INTERVAL_TIME, {
+                trailing: true,
+                leading: false,
+            }),
+            []
+        );
+
```
I saw in some places debounce effect looks good, but why we need this behaviour here? Why can't we send it immidiately?

NikitK-deriv commented on packages/bot-web-ui/src/components/quick-strategy/selects/contract-type.tsx:
```diff
@@ -99,6 +121,7 @@ const ContractTypes: React.FC<TContractTypes> = observer(({ fullWidth = false, n
                             onItemSelection={(item: TItem) => {
                                 if ((item as TContractTypesItem)?.value) {
                                     handleChange((item as TContractTypesItem)?.value as string);
+                                    sendTradeTypeToRudderStack(item);
```
```suggestion
                                    sendTradeTypeToRudderStack(item.text);
```

NikitK-deriv reviewed (changes requested):
Please provide some params as well 🙏 If not possible, let's discuss
event: ce_bot_dashboard_form, param: preview_mode

event: ce_bot_quick_strategy_form, params: strategy_switcher_mode, manual_parameter_input, action: switch_strategy_mode

Brilliant work 🔥

rupato-deriv commented on packages/bot-web-ui/src/components/dashboard/bot-builder/bot-builder.tsx:
```diff
@@ -18,6 +20,22 @@ const BotBuilder = observer(() => {
+
+    React.useEffect(() => {
+        if (active_tab === DBOT_TABS.BOT_BUILDER) {
+            trackRudderStackForBotBuilder('open');
```
done ✅

rupato-deriv commented on packages/bot-web-ui/src/components/dashboard/dashboard-component/cards.tsx:
```diff
@@ -35,6 +37,32 @@ const Cards = observer(({ is_mobile, has_dashboard_strategies }: TCardProps) =>
     const { handleFileChange, loadFileFromLocal } = load_modal;
     const { setFormVisibility } = quick_strategy;
 
+    //this is to check on click of which icon the user has come to bot builder
+    const sentToRudderStackTabChange = (type: string) => {
+        Analytics.trackEvent('ce_bot_builder_form', {
```
removed action

NikitK-deriv commented on packages/bot-web-ui/src/components/dashboard/dashboard-component/cards.tsx:
```diff
@@ -35,6 +37,32 @@ const Cards = observer(({ is_mobile, has_dashboard_strategies }: TCardProps) =>
+    const sentToRudderStackTabChange = (type: string) => {
+        Analytics.trackEvent('ce_bot_builder_form', {
+            shortcut_name: type,
+            form_source: 'bot_dashboard_form-shortcut',
```
does this proper source for that action?

NikitK-deriv replied:
let's remove form_source here then

rupato-deriv replied:
yes i have already added action there 
![Uploading image.png…]()

NikitK-deriv commented on packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/delete-dialog.tsx:
```diff
@@ -58,10 +59,21 @@ const DeleteDialog = observer(() => {
+    //this is to check after click of yes ir no on delete dailog send event to rudderstck
+    const sentToRudderStack = (param: string) => {
+        Analytics.trackEvent('ce_bot_builder_form', {
+            delete_popup_respond: param,
+            form_source: 'ce_bot_dashboard_form',
```
please add an action

NikitK-deriv replied:
t might be our mistake, let's keep it then 🙏

NikitK-deriv commented on packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent-workspace.tsx:
```diff
@@ -24,6 +25,8 @@ type TRecentWorkspace = {
 };
 
 const RecentWorkspace = observer(({ workspace, index }: TRecentWorkspace) => {
+    const { ui } = useStore();
+    const { is_mobile } = ui;
```
looks like you're not using it anywhere
or it's some kind of refactoring in parallel?

rupato-deriv replied:
i just wanted to remove isMobile from shared and use the is_mobile from ui store 🙏

NikitK-deriv replied:
I see, great 💪

NikitK-deriv commented on packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent.tsx:
```diff
@@ -2,33 +2,57 @@ import React from 'react';
+            const getStratagiesForRudderStack = async () => {
+                const recent_strategies = await getSavedWorkspaces();
+                Analytics.trackEvent('ce_bot_dashboard_form', {
+                    bot_last_modified_time: recent_strategies?.[0]?.timestamp,
+                    form_source: 'bot_header_form',
```
please add an action, like `edit_your_bot`

NikitK-deriv commented on packages/bot-web-ui/src/components/dashboard/tutorial-tab/common/search-input/index.tsx:
```diff
@@ -2,17 +2,39 @@ import { observer } from '@deriv/stores';
 import React from 'react';
+import { Analytics } from '@deriv/analytics';
+import debounce from 'lodash.debounce';
+import { DEBOUNCE_INTERVAL_TIME } from 'Constants/bot-contents';
```
should this folder has name uppercased?

rupato-deriv replied:
actually we use constants as caps in our package 
<img width="964" alt="image" src="https://github.com/binary-com/deriv-app/assets/97010868/5e1c1338-c913-442c-9de3-4b7de9c54ddb">

NikitK-deriv replied:
haha, that...suddenly. but okay, thanks for explanation

NikitK-deriv commented on packages/bot-web-ui/src/components/dashboard/bot-builder/toolbox/toolbox.tsx:
```diff
@@ -39,9 +41,17 @@ const Toolbox = observer(() => {
+    const sendToRudderStackOnQuickStrategyIconClick = () => {
+        Analytics.trackEvent('ce_bot_quick_strategy_form', {
+            action: 'open',
+            form_source: 'bot_builder_form',
```
Leave it then. I have to refactor this key everywhere later

rupato-deriv replied:
@NikitK-deriv just for the record we are using 1 route and we dont use nested routes on deriv-app

NikitK-deriv reviewed (approved):
LGTM

ali-hosseini-deriv commented:
@rupato-deriv the only concern that i have is that we added lots of things to the stores. having said that we are going remove the mobx stores in the future. I'm thinking that is it possible to not add those stuff in the stores and instead have them as hooks that are not using the stores directly?

rupato-deriv commented:
> @rupato-deriv the only concern that i have is that we added lots of things to the stores. having said that we are going remove the mobx stores in the future. I'm thinking that is it possible to not add those stuff in the stores and instead have them as hooks that are not using the stores directly?

@ali-hosseini-deriv Here i have added just 2 methods to the store. Maybe we can discuss it. 🙏
